IMS/Infrastructure/Models/InventoryRepository.cs

```diff
@@ -0,0 +1,61 @@
+﻿using IMS.Models;
+using Microsoft.EntityFrameworkCore;
+
+namespace IMS.Models
+{
+    public class InventoryRepository : IInventoryRepository
+    {
+        private readonly IMSDbContext _imsDbContext;
+
+        public InventoryRepository(IMSDbContext imsDbContext)
+        {
+            _imsDbContext = imsDbContext;
+        }
+
+        public async Task<IEnumerable<Inventory>> GetInventoryAsync(int page, int limit)
+        {
+            return await _imsDbContext.Inventory
```

Comment:
Add includes to eagerly load navigation properties:

```suggestion
            return await _imsDbContext.Inventory
                 .Include(c => c.Product)
                 .Include(c => c.Location
```